business_requirement_deliverable_project/tests/test_br.py

```diff
@@ -0,0 +1,286 @@
+
+        # Product Created A, B, C, D
+        self.ProductObj = self.env['product.product']
+        self.productA = self.ProductObj.create(
+            {'name': 'Product A', 'uom_id': self.uom_hours.id,
+                'uom_po_id': self.uom_hours.id,
+                'standard_price': 450})
+        self.productB = self.ProductObj.create(
+            {'name': 'Product B', 'uom_id': self.uom_hours.id,
+                'uom_po_id': self.uom_hours.id,
+                'standard_price': 550})
+        self.productC = self.ProductObj.create(
+            {'name': 'Product C', 'uom_id': self.uom_days.id,
+                'uom_po_id': self.uom_days.id,
+                'standard_price': 650})
+        self.productD = self.ProductObj.create(
+            {'name': 'Product D', 'uom_id': self.uom_kg.id,
+                'uom_po_id': self.uom_kg.id,
+                'standard_price': 750})
+
+        vals = {
+            'description': 'test',
+            'project_id': self.projectA.id,
+            'deliverable_lines': [
+                (0, 0, {'name': 'deliverable line1', 'qty': 1.0,
+                        'unit_price': 900, 'uom_id': 1,
+                        'resource_ids': [
+                            (0, 0, {
+                                'name': 'Resource Line1',
+                                'product_id': self.productA.id,
+                                'qty': 100,
+                                'uom_id': self.uom_hours.id,
+                                'resource_type': 'task',
+                            }),
+                            (0, 0, {
+                                'name': 'Resource Line1',
+                                'product_id': self.productB.id,
+                                'qty': 100,
+                                'uom_id': self.uom_hours.id,
+                                'resource_type': 'task',
+                            })
+                        ]
+                        }),
+                (0, 0, {'name': 'deliverable line2', 'qty': 1.0,
+                        'unit_price': 1100, 'uom_id': 1}),
+                (0, 0, {'name': 'deliverable line3', 'qty': 1.0,
+                        'unit_price': 1300, 'uom_id': 1}),
+                (0, 0, {'name': 'deliverable line4', 'qty': 1.0,
+                        'unit_price': 1500, 'uom_id': 1,
+                        }),
+            ],
+        }
+
+        self.brA = self.env['business.requirement'].create(vals)
+        self.brB = self.env['business.requirement'].create(vals)
+        self.brC = self.env['business.requirement'].create(vals)
+
+    def test_br_state(self):
+        # test when state=draft
+        self.brA.state = 'draft'
+        self.brB.state = 'draft'
+        self.brC.state = 'draft'
+        try:
+            action = self.projectA.generate_project_wizard()
+        except Exception, e:
+            action = False
+            self.assertEqual(action, False)
+            if type(e) == osv.except_osv:
```

Comment:
if you want to check the case of fail... you could use `self.assertRaiseRegex`
[See example](https://github.com/OCA/rma/blob/c60a144c3c509caf29a9d88aaf0c934fc39a2bfc/crm_rma_lot_mass_return/tests/test_constrains.py#L40)